### ISSUE TITLE
Reading `__v` from the outside of the class, ty doesn't give any error while Python interpreter gives error

### ISSUE THREAD

**Issue by hyperkai** [astral-sh/ty#2289]
### Summary

*Memo:
- ty check
- ty 0.0.7
- python test.py
- Python 3.14.0
- Windows 11

Reading `__v` from the outside of the class, ty doesn't give any error as shown below:

*Memo:
- A double leading underscore(__abc) can make attributes private in a class.

```python
class Cls:
    __v = 100

cls: Cls = Cls()

print(cls.__v) # No error
```

Actually, Python interpreter gives the error properly as shown below:

> AttributeError: 'Cls' object has no attribute '__v'

So, ty should give an error like Python interpreter gives as shown below:

> error: 'Cls' object has no attribute '__v'

### Version

ty 0.0.7

**Comment by AlexWaygood** [comment]
Thanks! Please see #645
